rustainers/src/runner/inner.rs

```diff
@@ -287,6 +331,82 @@ pub(crate) trait InnerRunner: Display + Debug + Send + Sync {
         Ok(())
     }
 
+    fn get_docker_host(&self) -> Option<String> {
+        env::var("DOCKER_HOST").ok()
+    }
+
+    #[tracing::instrument(skip(self),fields(runner = %self))]
+    async fn find_host_network(&self) -> Result<Option<Network>, ContainerError> {
+        // If we're docker in docker running on a custom network, we need to inherit the
+        // network settings, so we can access the resulting container.
+        let docker_host = self.host().await?;
+        let custom_networks = self.list_custom_networks().await?;
+        let mut networks = vec![];
+        for network in custom_networks {
+            let network_configs = self.list_network_config(network.id).await?;
+            networks.extend(
+                network_configs
+                    .into_iter()
+                    .filter_map(|x| {
+                        x.subnet.map(|x| {
+                            x.contains(IpAddr::V4(docker_host.0))
+                                .then(|| Network::Custom(network.name.clone()))
+                                .ok_or(0)
+                        })
+                    })
+                    // Chained filtermaps is not ideal
+                    .filter_map(std::result::Result::ok)
+                    .collect::<Vec<Network>>(),
+            );
+        }
+        if let [network] = &networks[..] {
+            Ok(Some(network.clone()))
+        } else {
+            Ok(None)
+        }
+    }
+
+    #[tracing::instrument(skip(self),fields(runner = %self))]
+    async fn host(&self) -> Result<Ip, ContainerError> {
+        if self.is_inside_container() {
+            self.default_gateway_ip().await
+        } else {
+            Ok(Ip(Ipv4Addr::LOCALHOST))
+        }
+    }
+
+    #[tracing::instrument(skip(self), fields(runner = %self))]
+    async fn default_gateway_ip(&self) -> Result<Ip, ContainerError> {
+        let hostname = env::var("HOSTNAME")?;
+        let host_id = hostname.parse::<ContainerId>()?;
+        let networks = self.inspect_networks(host_id).await?;
+        // Filter when values are defined
+        let networks = networks
+            .iter()
+            .filter_map(|(_, network)| match (network.id, network.gateway) {
```

Comment:
🤌 **nitpick**: you can avoid the match with `network.zip(network.gateway)`.
I've changed this method in the commit that fix podman